jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/resource/PathResource.java

```diff
@@ -263,10 +271,45 @@ public Path getPath()
         return path;
     }
 
+    public List<Resource> list()
+    {
+        if (!isDirectory())
+            return List.of(); // empty
+
+        try (Stream<Path> dirStream = Files.list(getPath()))
+        {
+            return dirStream.map(PathResource::new).collect(Collectors.toCollection(ArrayList::new));
+        }
+        catch (DirectoryIteratorException e)
+        {
+            LOG.debug("Directory list failure", e);
+        }
+        catch (IOException e)
+        {
+            LOG.debug("Directory list access failure", e);
+        }
+        return List.of(); // empty
+    }
+
     @Override
     public String getName()
     {
-        return path.toAbsolutePath().toString();
+        Path abs = path;
+        // If a "jar:file:" based path, we should normalize here, as the toAbsolutePath() does not resolve "/../" style segments in all cases
+        if ("jar".equalsIgnoreCase(path.toUri().getScheme()))
```

Comment:
Shouldn't this logic be in `MountedPathResource`?

Reply:
Moved. Done.